src/flash-source-buffer.js

```diff
@@ -382,17 +392,26 @@ export default class FlashSourceBuffer extends videojs.EventTarget {
 
       if (startTag.pts >= targetPts) {
         filteredVideoTags.push(startTag);
-      } else if (startTag.keyFrame) {
-        let nextIndex = startIndex + 1;
+      } else if (startTag.metaDataTag || startTag.keyFrame) {
+        let keyFrameIndex = getKeyFrameIndex(startTag, startIndex);
```

Comment:
Minor, but it seems a bit odd to me that we get the key frame index, but it's only the key frame index sometimes (from the call to verify below).